### PR TITLE
set start end dates for agreement

### PR DESCRIPTION
**Pull request by ntrncic**
fixes unicef/etools-issues#478

### CODE REVIEW

File: EquiTrack/partners/models.py [hunk 1]
```diff
@@ -1088,6 +1088,18 @@ def save(self, **kwargs):
...
         self.update_related_interventions(oldself)
 
+        if self.agreement_type in [Agreement.PCA]:
+            # set start date
+            if self.signed_by_partner_date and self.signed_by_unicef_date:
+                self.start = self.signed_by_unicef_date \
+                    if self.signed_by_unicef_date > self.signed_by_partner_date else self.signed_by_partner_date
+
+            # set end date
+            if self.country_programme:
```
Comment (robertavram) [inline EquiTrack/partners/models.py]: until this is signed, there's no need to save the end date. under the previous if (` if self.signed_by_partner_date and self.signed_by_unicef_date:`) you can nest this `self.end = self.country_programme.to_date` but assert that country programme exists first

Comment (ntrncic) [inline EquiTrack/partners/models.py]: Done

File: EquiTrack/partners/tests/test_models.py [hunk 2]
```diff
@@ -172,22 +172,31 @@ def setUp(self):
         self.partner_organization = PartnerOrganization.objects.create(
             name="Partner Org 1",
         )
+        self.cp = CountryProgramme.objects.create(
+            name="CP 1",
+            wbs="/A0/",
```
Comment (robertavram) [inline EquiTrack/partners/tests/test_models.py]: give it a proper WBS. CP will label this special if 'A' is not the 5th character

**Comment by robertavram** [comment]
looks good.. just one comment

File: EquiTrack/partners/tests/test_models.py (hunk 2 above)
Comment (ntrncic) [inline EquiTrack/partners/tests/test_models.py]: Done!

**Pull request merged**